lib/util/sdp/issue8329.js

```diff
@@ -0,0 +1,205 @@
+'use strict';
+
+var RTCSessionDescription = require('@twilio/webrtc').RTCSessionDescription;
+
+var createPtToCodecName = require('./').createPtToCodecName;
+var getMediaSections = require('./').getMediaSections;
+
+/**
+ * An RTX payload type
+ * @typedef {PT} RtxPT
+ */
+
+/**
+ * A non-RTX payload type
+ * @typedef {PT} NonRtxPT
+ */
+
+/**
+ * A Set with at least one element
+ * @typedef {Set} NonEmptySet
+ */
+
+/**
+ * Apply the workaround for Issue 8329 to an RTCSessionDescriptionInit.
+ * @param {RTCSessionDescriptionInit} description
+ * @returns {RTCSessionDescription} newDescription
+ */
+function workaround(description) {
+  var sdp;
+  if (description.type === 'offer' || description.type === 'answer' || description.type === 'pranswer') {
+    sdp = sdpWorkaround(description.sdp);
+  }
+  var descriptionInit = { type: description.type };
+  if (sdp) {
+    descriptionInit.sdp = sdp;
+  }
+  return new RTCSessionDescription(descriptionInit);
+}
+
+/**
+ * @param {string} sdp
+ * @returns {string} newSdp
+ */
+function sdpWorkaround(sdp) {
+  var mediaSections = getMediaSections(sdp);
+  var session = sdp.split('\r\nm=')[0];
+  return [session]
+    .concat(mediaSections.map(mediaSectionWorkaround))
+    .join('\r\n');
+}
+
+/**
+ * @param {string} mediaSection
+ * @returns {string} newMediaSection
+ */
+function mediaSectionWorkaround(mediaSection) {
+  var ptToCodecName = createPtToCodecName(mediaSection);
+  var codecNameToPts = createCodecNameToPts(ptToCodecName);
+  var rtxPts = codecNameToPts.get('rtx') || new Set();
+
+  var invalidRtxPts = new Set();
+  var rtxPtToAssociatedPt = createRtxPtToAssociatedPt(
+    mediaSection, ptToCodecName, rtxPts, invalidRtxPts);
+  var associatedPtToRtxPt = createAssociatedPtToRtxPt(
+    rtxPtToAssociatedPt, invalidRtxPts);
+
+  var unassociatedRtxPts = Array.from(invalidRtxPts);
+
+  // NOTE(mroberts): We normalize to lowercase.
+  var knownCodecNames = ['h264', 'vp8', 'vp9'];
+  var unassociatedPts = knownCodecNames.reduce(function(unassociatedPts, codecName) {
+    var pts = codecNameToPts.get(codecName) || new Set();
+    return Array.from(pts).reduce(function(unassociatedPts, pt) {
+      return associatedPtToRtxPt.has(pt)
+        ? unassociatedPts
+        : unassociatedPts.add(pt);
```

Comment:
@markandrus Are you sure `Set#add()` returns the `Set`?

Reply:
Yes https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add